### DISCUSSION TITLE
Dreambooth runs with 8000 steps despite 20x200 setup

### DISCUSSION THREAD

**Discussion by giteeeeee** [d8ahazard/sd_dreambooth_extension#841] [Q&A]
I started training using a setup of 20 instance images + Training Steps Per Image (Epochs) = 200, batch size and GA to 1.

From my understanding such setup should give 4000 training steps, but dreambooth started to train with 8000 steps:

![screenshot 222](https://user-images.githubusercontent.com/89521447/214205159-c78a5a76-f58e-49b2-9747-9d50c718ee3e.jpg)

By the picture, can anyone tell me what's wrong? Thanks.

**Comment by d8ahazard** [comment]
Class images also count as training images. So, if you are using class
images, the total number of training steps doubles. This will obviously
vary, if, say, you have some concepts with class images, and others without.

**Reply by giteeeeee** [reply]
Hey thanks for helping!

I just noticed that I did set my text encoder ratio to 1. Maybe it's training unet 4000 steps + text encoder 4000 steps = 8000 steps?

If this is the case, should I set "training Steps Per Image (Epochs)" from 200 to 100 instead?

Or, like you said, if my class images of 80 are counted towards training images, shouldn't the training step be (20+80) x 200 = 20000?

**Comment by d8ahazard** [comment]
No. Because for each epoch, only one class images used in a pair with an
instance image. All of your class images will be used, just not every class
image every badge. This way  training stays balanced.

**Reply by giteeeeee** [reply]
Understood.
So I came from Shivam's repo, is
"20 instance images + **100** Steps Per Image + reg img"
or
"20 instance images + **200** Steps Per Image + reg img" in your repo
equivalent to
"20 instance images + 4000 training steps + reg img" in his repo?
